### DISCUSSION TITLE
Customize search result summary

### DISCUSSION THREAD

**Discussion by jtatum** [squidfunk/mkdocs-material#3012] [Q&A] [answered]
### Contribution guidelines

- [X] I've read the [contribution guidelines](https://github.com/squidfunk/mkdocs-material/blob/master/CONTRIBUTING.md) and wholeheartedly agree

### I want to suggest an idea and checked that ...

- [X] ... to my best knowledge, my idea wouldn't break something for other users
- [X] ... the documentation does not mention anything about my idea
- [X] ... there are no open or closed issues that are related to my idea

### Description

I know there are changes to search result summaries coming with #2976, but I'm wondering what I can do now to change the search result summary. Is there a way to override the function that generates or returns these, or some front matter I can use to set a static summary?

### Use Cases

The data we have in mkdocs is highly structured, and the summaries are very low quality, with most failing to even show the page title.

**Comment by squidfunk** [comment]
Converting to discussion, since this is essentially a question.

**Comment by squidfunk** [comment, marked as answer]
Currently, the search result text is just truncated. You could either customize the `truncate` function:

https://github.com/squidfunk/mkdocs-material/blob/f23f738bc0f94c50ced84b91e7cde7fb740fb912/src/assets/javascripts/utilities/string/index.ts#L27-L47

... or alter the search result rendering function:

https://github.com/squidfunk/mkdocs-material/blob/f23f738bc0f94c50ced84b91e7cde7fb740fb912/src/assets/javascripts/templates/search/index.tsx#L89-L91

On a side note, the new search should be ready shortly. Search summaries are really hard to get right for all languages (not just for space-separated, sentence-based ones). I've wanted to improve search summaries for a long, long time, but only had the idea how to do it quite recently. If you really can't wait, you can fork the theme and change the behavior of the sections I linked above 😊
